torchtext/datasets/text_classification.py

```diff
@@ -41,23 +41,29 @@ def _csv_iterator(data_path, ngrams, yield_cls=False):
                 yield ngrams_iterator(tokens, ngrams)
 
 
-def _create_data_from_iterator(vocab, iterator, include_unk):
-    data = []
-    labels = []
-    with tqdm(unit_scale=0, unit='lines') as t:
-        for cls, tokens in iterator:
-            if include_unk:
-                tokens = torch.tensor([vocab[token] for token in tokens])
-            else:
-                token_ids = list(filter(lambda x: x is not Vocab.UNK, [vocab[token]
-                                        for token in tokens]))
-                tokens = torch.tensor(token_ids)
-            if len(tokens) == 0:
-                logging.info('Row contains no tokens.')
-            data.append((cls, tokens))
-            labels.append(cls)
-            t.update(1)
-    return data, set(labels)
+def _create_data_from_iterator(vocab, iterator, removed_tokens):
+    for cls, tokens in iterator:
+        yield cls, iter(map(lambda x: vocab[x],
+                        filter(lambda x: x not in removed_tokens, tokens)))
+
+
+def _imdb_iterator(key, extracted_files, tokenizer, ngrams, yield_cls=False):
+    for fname in extracted_files:
+        if 'urls' in fname:
+            continue
+        elif key in fname:
+            if 'pos' in fname:
+                with io.open(fname, encoding="utf8") as f:
```

Comment:
You could collapse this into one by saying

```label = 0 if 'pos' in fname else 1```
And then do 
```
 with io.open(fname, encoding="utf8") as f:
     token_iterator = ngrams_iterator(tokenizer(f.read()), ngrams)
     if yield_cls:
         yield int(label), token_iteratator
     else:
         yield token_iterator
```

Are you sure the explicit cast via ```int``` is necessary here?

Reply:
Label is actually the index. When applying the dataset to the model, labels are used as the target so probably a floating value is more proper.